lldb/source/Plugins/ExpressionParser/Clang/ClangASTImporter.cpp

```diff
@@ -1411,3 +1414,35 @@ clang::Decl *
 ClangASTImporter::ASTImporterDelegate::GetOriginalDecl(clang::Decl *To) {
   return m_main.GetDeclOrigin(To).decl;
 }
+
+void ClangASTImporter::ASTImporterDelegate::UpdateImportProgress(
```

Comment:
The assertions in this code are going to blow up only in development builds. In release builds, you'll fall through the first assertion, and return because you try to cast `nullptr` to a NamedDecl (which will give you `nullptr`) and the function will fail silently. The second assertion will fail if `from_ast` or `to_ast` is `nullptr`, but the function doesn't return early there. It will call `getDisplayName` on `from_ast` and `to_ast`, one of which will crash anyway. Maybe we should consider an early return here too? Or maybe the function should return an `llvm::Error`?

Reply:
These asserts should never ever actually occur unless something about TypeSystemClang gets redesigned, at which point the intention was that these would flag in development builds/tests. But definitely valid question. For the cases where a `nullptr` would occur I did try to be consistent with early returns. I usually try to express the preconditions of the function in terms of asserts, but in this case I don't have strong opinion, I think doing an early return would make sense here. I remember some time last year @DavidSpickett proposed an assert macro that would fire in debug builds but early-return in release. Can't find that atm, not sure if it ever landed.

Reply:
> cast nullptr to a NamedDecl (which will give you nullptr) 

Actually, `dyn_cast` will abort if it gets a null pointer (`dyn_cast_or_null` is the version that forwards nullness) -- which makes this assert unnecessary. However, I think this is a very good example of an `assert` -- checking for things that "should not happen" without introducing overhead in the production codepath.